src/scenes/gamebanana.lua

```diff
@@ -389,10 +393,11 @@ function scene.downloadSearchEntries(query)
     return data, msg
 end
 
-function scene.downloadSortedEntries(page, sort, itemtypeFilterType, itemtypeFilterValue)
+function scene.downloadSortedEntries(page, sort, itemtypeFilter)
     local url = "https://max480-random-stuff.appspot.com/celeste/gamebanana-list?" ..
         (sort ~= "" and "sort=" .. sort .. "&" or "") ..
-        (itemtypeFilterValue ~= "" and itemtypeFilterType .. "=" .. itemtypeFilterValue .. "&" or "") ..
+        (itemtypeFilter.itemtype and "type=" .. itemtypeFilter.itemtype .. "&" or "") ..
+        (itemtypeFilter.categoryid and "category=" .. itemtypeFilter.categoryid .. "&" or "") ..
```

Comment:
I would personally prefer some format strings here, or a query "builder" util, but I'm not too sure how much this kind of logic is used in Olympus. Nothing wrong with this, just a bit hard to read and potentially maintain in the future.